daemon/src/maker_cfd_actor.rs

```diff
@@ -300,7 +298,107 @@ impl MakerCfdActor {
 
         Ok(())
     }
+
+    async fn handle_accept_order(&mut self, msg: AcceptOrder) -> Result<()> {
+        tracing::debug!(%msg.order_id, "Maker accepts an order" );
+
+        let mut conn = self.db.acquire().await?;
+
+        // Validate if order is still valid
+        let cfd = load_cfd_by_order_id(msg.order_id, &mut conn).await?;
+
+        let taker_id = match cfd {
+            Cfd {
+                state: CfdState::IncomingOrderRequest { taker_id, .. },
+                ..
+            } => taker_id,
+            _ => {
+                tracing::warn!("Order is in invalid state. Ignoring trying to accept it.");
+                anyhow::bail!("Order is in invalid state. Ignoring trying to accept it.")
+            }
+        };
+
+        // Update order in db
+        insert_new_cfd_state_by_order_id(
+            cfd.order.id,
+            CfdState::Accepted {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+            &mut conn,
+        )
+        .await
+        .unwrap();
+
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::TakerMessage {
+                taker_id,
+                command: TakerCommand::NotifyOrderAccepted { id: msg.order_id },
+            })
+            .await?;
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::BroadcastOrder(None))
+            .await?;
+        self.order_feed_sender.send(None)?;
```

Comment:
As per the other comment, this needs to happen together with setting `current_order_id` to `None`.